storey/sources.py

```diff
@@ -80,27 +81,32 @@ class FlowController(FlowControllerBase):
     To be used from a synchronous context.
     """
 
-    def __init__(self, emit_fn, await_termination_fn, key_field: Optional[str] = None,
+    def __init__(self, emit_fn, await_termination_fn, has_complete_in_flow, key_field: Optional[str] = None,
```

Comment:
Here it makes sense to call it return_awaitable_result, not has_complete_in_flow.